R/matchit.R

```diff
@@ -1,20 +1,29 @@
+
+# utility function for MatchIt::matchit() to check data inputs
+check_data_missingness <- function(formula, data) {
+  if (!inherits(formula, "formula")) { formula <- formula(formula) }
+  if(is.null(data)) stop("Dataframe must be specified",call.=FALSE)
+  if(!is.data.frame(data)){
+    stop("Data must be a dataframe",call.=FALSE)}
+  
+  varnames <- all.vars(formula)
+  var_idx <- which(names(data) %in% varnames)
+  if (length(varnames) == 0) stop("formula does not contain variables")
+  if ( any(!(varnames %in% names(data))) ) stop("formula variables do not exist in data.")
+  if (any(apply(data[, var_idx], 2, function(j) sum(is.na(j) || is.nan(j))) > 0))
```

Comment:
This line only works if all variables are contained in `var_idx` but in some cases (like nearest neighbor matching with exact constraint) other inputs contain variables and so we need to check the existence of missing values in those variables too.